src/js/controller/FramesListController.js

```diff
@@ -87,7 +87,11 @@
     } else if (action === ACTION.SELECT) {
       this.piskelController.setCurrentFrameIndex(index);
     } else if (action === ACTION.NEW_FRAME) {
-      this.piskelController.addFrame();
+      // Publish an event that the user selected add new frame.
+      $.publish(Events.ADD_NEW_FRAME_CLICKED);
+      // Disable adding a blank frame and idead use the PiskelApi
+      // to cause an action.
+      //this.piskelController.addFrame();
```

Comment:
The more I think about this, the more I'd like to see it be more flexible; as if the service should offer each attached client an opportunity to handle "new frame" and proceed to its own logic if all clients refuse.  The 'old behavior if no custom handler' version is probably the MVP for this, and would be cheap-ish to implement now.  Thoughts?